test/component_server_test.dart

```diff
@@ -55,6 +55,7 @@ void main() {
 
       // Simulate un-mounting the component
       component.componentWillUnmount();
+      await component.didDispose;
```

Comment:
dispose is already getting called as a side effect of componentWillUnmount - it just wasn't happening fast enough for the subsequent test validations to work as expected without this await